sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -113,48 +120,74 @@ def evaluate(
 
     # Run temporal cross-validation.
     for i, (train, test) in enumerate(cv.split(y)):
+
+        # set default result values in case estimator fitting fails
+        score = error_score
+        fit_time = np.nan
+        pred_time = np.nan
+        cutoff = np.nan
+        y_pred = np.nan
+
         # split data
         y_train, y_test, X_train, X_test = _split(y, X, train, test, cv.fh)
 
         # create forecasting horizon
         fh = ForecastingHorizon(y_test.index, is_relative=False)
 
-        # fit/update
-        start_fit = time.perf_counter()
-        if i == 0 or strategy == "refit":
-            forecaster = forecaster.clone()
-            forecaster.fit(y_train, X_train, fh=fh)
-
-        else:  # if strategy in ["update", "no-update_params"]:
-            update_params = strategy == "update"
-            forecaster.update(y_train, X_train, update_params=update_params)
-        fit_time = time.perf_counter() - start_fit
-
-        pred_type = {
-            "pred_quantiles": "forecaster.predict_quantiles",
-            "pred_intervals": "forecaster.predict_interval",
-            "pred_proba": "forecaster.predict_proba",
-            None: "forecaster.predict",
-        }
-        # predict
-        start_pred = time.perf_counter()
-
-        if hasattr(scoring, "metric_args"):
-            metric_args = scoring.metric_args
-
         try:
-            scitype = scoring.get_tag("scitype:y_pred")
-        except ValueError:
-            # If no scitype exists then metric is not proba and no args needed
-            scitype = None
-            metric_args = {}
-
-        y_pred = eval(pred_type[scitype])(fh, X_test, **metric_args)
-
-        pred_time = time.perf_counter() - start_pred
-
-        # score
-        score = scoring(y_test, y_pred, y_train=y_train)
+            # fit/update
+            start_fit = time.perf_counter()
+            if i == 0 or strategy == "refit":
+                forecaster = forecaster.clone()
+                forecaster.fit(y_train, X_train, fh=fh)
+
+            else:  # if strategy in ["update", "no-update_params"]:
+                update_params = strategy == "update"
+                forecaster.update(y_train, X_train, update_params=update_params)
+            fit_time = time.perf_counter() - start_fit
+
+            pred_type = {
+                "pred_quantiles": "forecaster.predict_quantiles",
+                "pred_intervals": "forecaster.predict_interval",
+                "pred_proba": "forecaster.predict_proba",
+                None: "forecaster.predict",
+            }
+            # predict
+            start_pred = time.perf_counter()
+
+            if hasattr(scoring, "metric_args"):
+                metric_args = scoring.metric_args
+
+            try:
+                scitype = scoring.get_tag("scitype:y_pred")
```

Comment:
not from this PR, but this could be shortened by using the second argument?

Reply:
which second argument?

Reply:
of `get_tag`, named `tag_value_default`

Reply:
hm, pls make code suggestion. Otherwise its anyway not related to this PR

Reply:
I would replace line 158-166 by:
```python
scitype = scoring.get_tag("scitype:y_pred", None)

if hasattr(scoring, "metric_args") and scitype is not None:
    metric_args = scoring.metric_args
else:
    metric_args = {}
```

Reply:
I tried this but it was failing. Lets just finish this PR.